packages/core/src/package/postDeployers/FHTEnabler.ts

```diff
@@ -0,0 +1,99 @@
+import { Schema } from 'jsforce';
+import CustomFieldFetcher from '../../metadata/CustomFieldFetcher';
+import SFPOrg from '../../org/SFPOrg';
+import path from 'path';
+const { XMLBuilder } = require('fast-xml-parser');
+
+const QUERY_BODY =
+    'SELECT QualifiedApiName, EntityDefinitionId  FROM FieldDefinition WHERE IsFieldHistoryTracked = true AND EntityDefinitionId IN ';
+
+export default class FHTEnabler implements PostDeployer {
+    public async isEnabled(sfpPackage: SfpPackage, conn: Connection<Schema>, logger: Logger): Promise<boolean> {
+        if (
+            sfpPackage['isFHTFieldFound'] &&
+            (sfpPackage.packageDescriptor.enableFHT == undefined || sfpPackage.packageDescriptor.enableFHT == true)
+        ) {
+            return true;
+        }
+    }
+
+    public async gatherPostDeploymentComponents(
+        sfpPackage: SfpPackage,
+        componentSet: ComponentSet,
+        conn: Connection,
+        logger: Logger
+    ): Promise<{ location: string; componentSet: ComponentSet }> {
+        //First retrieve all objects/fields  of interest from the package
+        let objList = [];
+        let fieldList = [];
+        Object.keys(sfpPackage['fhtFields']).forEach((key) => {
+            objList.push(`'${key}'`);
+            sfpPackage['fhtFields'][key].forEach((field) => fieldList.push(key + '.' + field));
+        });
+        //Now query all the fields for this object where FHT is already enabled
+        SFPLogger.log(`Gathering fields which are already  trackHistory enabled in the  target org....`, LoggerLevel.INFO, logger);
+        let fhtFieldsInOrg = await QueryHelper.query<{
+            QualifiedApiName: string;
+            EntityDefinitionId: string;
+            IsFieldHistoryTracked:boolean;
+        }>(QUERY_BODY + '(' + objList + ')', conn, true);
+
+
+        //Clear of the fiels that alread has FHT applied and keep a reduced filter
+        fhtFieldsInOrg.map((record) => {
+            let field = record.EntityDefinitionId + '.' + record.QualifiedApiName;
+            const index = fieldList.indexOf(field);
+            if (index > -1) {
+                fieldList.splice(index, 1);
+            }
+        });
+
+        if (fieldList.length > 0) {
+            //Now retrieve the fields from the org
+            let customFieldFetcher: CustomFieldFetcher = new CustomFieldFetcher(logger);
+            let sfpOrg = await SFPOrg.create({ connection: conn });
+            let fetchedCustomFields = await customFieldFetcher.getCustomFields(sfpOrg, fieldList);
+
+            //Modify the component set
+            for (const sourceComponent of fetchedCustomFields.components.getSourceComponents()) {
+                let sourceComponentXml = await sourceComponent.parseXml();
+
+                if (sourceComponent.type.name == registry.types.customobject.children.types.customfield.name) {
+                    sourceComponentXml['CustomField']['trackHistory'] = true;
```

Comment:
need to enable history tracking for the object as well (sourceObjectXml['CustomObject']['enableHistory']) in the case the object is not fht enabled in the org

Reply:
Will do that on a subsequent Pr, as this one exclusively looks at custom fields. Pulling custom object definition can be bit tricky